core/src/validator.rs

```diff
@@ -1501,6 +1525,8 @@ fn load_blockstore(
         blockstore_processor::ProcessOptions,
         BlockstoreRootScan,
         DroppedSlotsReceiver,
+        Option<EntryNotifierSender>,
+        Option<EntryNotifierService>,
```

Comment:
Because these are either both Some or both None (based on `initialize_entry_notifier_service()`), might be cleaner to:

1.  `EntryNotifierService` struct just owns the `sender`, exposed through an accessor
2.  `EntryNotifierService::new()` creates the sender and receiver: `let (entry_notification_sender, entry_notification_receiver) = unbounded();`
3. Instead of `initialize_entry_notifier_service()` we just do:

```
let entry_notifier_service = 
    entry_notifier.map(|entry_notifier| EntryNotifierService::new(entry_notifier, exit));
```

Reply:
The only reason I can think of not to do that is that it's a different pattern than all our other such services. But I've no objection to starting a new pattern.

Reply:
I worked this up, but I actually don't think it's cleaner. Wdyt @carllin ? https://github.com/solana-labs/solana/pull/31290/commits/5500c4a539f50f4fdea8ad12c8b9f86d5436350a

Reply:
I personally like this because everything is more encapsulated within `entry_notifier_service.rs`, and we don't have to pass the individual senders around